skfp/descriptors/topological.py

```diff
@@ -0,0 +1,210 @@
+from collections import Counter
+from typing import Optional
+
+import numpy as np
+from rdkit.Chem import GetDistanceMatrix, rdmolops, Mol
+
+
+def weiner_index(mol: Mol, distance_matrix: Optional[np.ndarray] = None) -> int:
+    """
+    Wiener Index (W).
+
+    The implementation uses RDKit to calculate the Weiner Index for molecular graphs.
+    The Weiner Index is computed as the sum of all pairwise distances in the molecular
+    graph's distance matrix.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecular graph for which the Wiener index is to be calculated.
+
+    distance_matrix : np.ndarray, optional
+        Precomputed distance matrix. If not provided, it will be calculated using RDKit.
+
+    References
+    ----------
+    .. [1] Chen, Ailian, Xianzhu Xiong, and Fenggen Lin.
```

Comment:
1. Why this reference? The original one is https://pubs.acs.org/doi/abs/10.1021/ja01193a005, second one could be e.g. https://www.sciencedirect.com/science/article/abs/pii/B9781898563761500068.
2. Missing backticks and underscore at the end, see e.g. https://github.com/scikit-fingerprints/scikit-fingerprints/blob/master/skfp/fingerprints/atom_pair.py#L119. This will create a hyperlink in the docs

Reply:
Second point applies to all docstrings